coderd/database/migrations/000025_allow_template_name_re_use.up.sql

```diff
@@ -0,0 +1,4 @@
+DROP INDEX idx_templates_name_lower;
+CREATE UNIQUE INDEX idx_templates_name_lower ON templates USING btree (lower(name)) WHERE deleted = false;
```

Comment:
Hmm, does this limitation actually make sense? Would we ever want to limit globally unique template names when we're already restricting them on a per-org basis?

I think both of these partial indexes could be consolidated into one:

```
DROP INDEX idx_templates_name_lower;
DROP INDEX templates_organization_id_name_idx;
CREATE INDEX templates_organization_id_name_idx ON templates (organization_id, lower(name)) WHERE deleted = false;
```

Thoughts @johnstcn @kylecarbs @Emyrk?

Reply:
I guess globally unique is a bit strange. So I think you are right.

I only see 1 partial index though. What do you mean by both?

Reply:
It's not visible in this PR since I didn't modify it, but it's there:

```
❯ rg 'deleted = false'
coderd/database/dump.sql
446:CREATE UNIQUE INDEX idx_templates_name_lower ON templates USING btree (lower((name)::text)) WHERE (deleted = false);
452:CREATE UNIQUE INDEX templates_organization_id_name_idx ON templates USING btree (organization_id, name) WHERE (deleted = false);
```

Reply:
Huh, well I agree only the 1 on `(organization_id, lower(name))` is needed.

Reply:
We definitely want cross-org unique name enforcement. I figure it wasn't an issue up till now because we only really have one org, but better fixed sooner than later.

> Error: your password cannot be the same as user `stevenmasley`

Reply:
@johnstcn for templates? So org `A` and org `B` cannot have the same named template `frontend-developer-template`?

Reply:
I really should have remembered to write "not" before "want".

Reply:
🥔, 🍅, it's all the same!